packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -0,0 +1,186 @@
+
+        FundingRateStoreInterface fundingRateStore = _getFundingRateStore();
+        FixedPoint.Unsigned memory _latestFundingRatePerSecondPerToken = fundingRateStore.getFundingRateForIdentifier(
+            identifier
+        );
+
+        FixedPoint.Unsigned memory effectiveFundingRateForPeriodPerToken;
+        (
+            cumulativeFundingRateMultiplier,
+            effectiveFundingRateForPeriodPerToken
+        ) = _calculateEffectiveFundingRatePerToken(
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken,
+            cumulativeFundingRateMultiplier
+        );
+        lastUpdateTime = currentTime;
+
+        emit NewFundingRate(
+            cumulativeFundingRateMultiplier.rawValue,
+            lastUpdateTime,
+            paymentPeriod,
+            _latestFundingRatePerSecondPerToken.rawValue,
+            effectiveFundingRateForPeriodPerToken.rawValue
+        );
+    }
+
+    function _calculateEffectiveFundingRatePerToken(
+        uint256 paymentPeriodSeconds,
+        FixedPoint.Unsigned memory fundingRatePerSecondPerToken,
+        FixedPoint.Unsigned memory feeMultiplier
+    ) internal pure returns (FixedPoint.Unsigned memory, FixedPoint.Unsigned memory) {
+        // Determine whether `fundingRatePerSecondPerToken` implies a negative or positive funding rate,
+        // and apply it over a pay period.
+        FixedPoint.Unsigned memory effectiveFundingRateForPeriodPerToken;
+        FixedPoint.Unsigned memory ONE = FixedPoint.fromUnscaledUint(1);
+
+        FixedPoint.Unsigned memory newFeeMultiplier;
+        if (fundingRatePerSecondPerToken.isEqual(ONE)) {
+            // If `fundingRatePerSecondPerToken` == 1, then maintain the current multiplier.
+
+            effectiveFundingRateForPeriodPerToken = ONE;
+            newFeeMultiplier = feeMultiplier;
+        } else if (fundingRatePerSecondPerToken.isGreaterThan(ONE)) {
+            // If `fundingRatePerSecondPerToken` > 1, then first scale the funding over the pay period:
+            // (`fundingRatePerSecondPerToken` - 1) * payPeriod = effectiveFundingRate.
+            // Next, multiply the current multipier by (1 + effectiveFundingRate).
+
+            effectiveFundingRateForPeriodPerToken = ONE.add(
+                fundingRatePerSecondPerToken.sub(ONE).mul(paymentPeriodSeconds)
+            );
+            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriodPerToken);
+        } else {
+            // If `fundingRatePerSecondPerToken` < 1, then first scale the funding over the pay period:
+            // (1 - `fundingRatePerSecondPerToken`) * payPeriod = effectiveFundingRate.
+            // Next, multiply the current multipier by (1 - effectiveFundingRate).
+
+            effectiveFundingRateForPeriodPerToken = ONE.sub(
+                ONE.sub(fundingRatePerSecondPerToken).mul(paymentPeriodSeconds)
+            );
+            newFeeMultiplier = feeMultiplier.mul(effectiveFundingRateForPeriodPerToken);
+        }
```

Comment:
I see what y'all were saying -- the lack of a SignedFixedPoint makes this more complicated than it needs to be. This doesn't necessarily need to change, but I think everything would be more intuitive if the rates came in from the store as positive and negative (centered around 0) and we were able to do this in a single case that handled both positive and negative (which would be pretty trivial if we could use negative numbers).

Probably something we should consider (but outside the scope of this PR IMO).

Reply:
Downside to this is that the users of the Store would be dealing with `int256`, including voters, unless we converted from `int256 -> FixedPoint` within the Store. WDYT?

Reply:
I think voters already deal with `int256` (see [here](https://github.com/UMAprotocol/protocol/blob/b3f243a37204bd71fa466f4da5e15754514b8063/packages/core/contracts/oracle/interfaces/OracleInterface.sol#L33)). Regardless, we could use `int256` or we could just build a `SignedFixedPoint` that's used everywhere (including in the interfaces). In general, though, the voters will have to return an `int256` that gets converted to a `FixedPoint` or a `SignedFixedPoint` in the contract.

Reply:
Thoughts?

Reply:
Hm this is a good point. I think we should have a couple discussions:
- pros/cons to building SignedFixedPoint
- whether voters should vote in an int256 or a SignedFixedPoint or a FixedPoint

Regardless I think we should push this decision to a subsequent PR. Wdyt? This PR mainly sets up the architecture of the FundingRateApplier but we should decide before we merge in #2084

Reply:
Thinking about this some more. I think there is an open question how the FundingRateStore should represent funding rates: as `int256`, `FixedPoint` or `SignedFixedPoint`. But, I think we should keep storing the `cumulativeFundingRateMultiplier` as a `FixedPoint` centered around 1. So, I think this PR can be unchanged.

Reply:
Yes 100%. We should have this discussion, but do it separately in its own issue/space. No changes needed right now.